modules/icgc-argo-workflows/prep/metrics/resources/usr/bin/rnaaln.py

```diff
@@ -0,0 +1,120 @@
+        'pct_coding_bases' : 'PCT_CODING_BASES',
+        'pct_utr_bases' : 'PCT_UTR_BASES',
+        'pct_intronic_bases' : 'PCT_INTRONIC_BASES',
+        'pct_intergenic_bases' : 'PCT_INTERGENIC_BASES',
+        'pct_mrna_bases' : 'PCT_MRNA_BASES',
+        'pct_usable_bases' : 'PCT_USABLE_BASES',
+        'pct_correct_strand_reads' : 'PCT_CORRECT_STRAND_READS',
+        'median_cv_coverage' : 'MEDIAN_CV_COVERAGE',
+        'median_5prime_to_3prime_bias' : 'MEDIAN_5PRIME_TO_3PRIME_BIAS'
+    }
+}
+
+fra2pct_fields = []
+integer_fields = ['paired_total', 'unpaired_total', 'total_reads', 'num_splices', 'num_annotated_splices', 'correct_strand_reads']
+
+def get_mqc_stats(multiqc, sampleId):
+    mqc_stats = {
+        'sample_id': sampleId,
+        'metrics': {}
+    }
+    for f in sorted(glob(multiqc+'/*.txt')):
+        for tool_metrics in tool_fieldmap.keys():
+            if f.endswith(tool_metrics+'.txt'):
+                with open(f, 'r') as fn: 
+                    mqc_stats[tool_metrics] = []
+                    reader = csv.DictReader(fn, delimiter="\t")
+                    for row in reader:
+                        if not sampleId in row.get('Sample'): continue
+                        mqc_stats[tool_metrics].append(row)
+                        for ftype in tool_fieldmap.keys():
+                            if not ftype == tool_metrics: continue
+                            for f1,f2 in tool_fieldmap[ftype].items():
+                                mqc_stats['metrics'][f1] = round(float(row.get(f2)), 4)
+
+    # convert the fraction to percentage for given fields
+    for fn in fra2pct_fields:
+        if not mqc_stats['metrics'].get(fn): 
+            print(f"Field '{fn}' not found in mqc_stats['metrics'] dictionary")
+            continue
+        new_value = round(float(mqc_stats['metrics'][fn]) * 100, 2)
+        mqc_stats['metrics'].update({
+            fn: new_value
+        })
+
+    # change type to integer for given fields
+    for fn in integer_fields:
+        if fn not in mqc_stats['metrics']: continue
+        new_value = int(mqc_stats['metrics'][fn])
+        mqc_stats['metrics'].update({
+            fn: new_value
+        })
+
+    return mqc_stats
+
+def main():
+    parser = argparse.ArgumentParser(description='Tool: prep_metrics')
+    parser.add_argument("-s", "--sampleId", dest="sampleId", required=True, help="Input sampleId", type=str)
+    parser.add_argument("-m", "--multiqc", dest="multiqc", required=True, help="multiqc files folder", type=str, nargs="+")
+
+    args = parser.parse_args()
+
+    # get tool_specific & aggregated metrics from multiqc
+    mqc_stats = {}
+    for fn in sorted(args.multiqc):
+        mqc_stats = get_mqc_stats(fn, args.sampleId)
+
+    mqc_stats_updated = {k: v for k, v in mqc_stats.items() if v}
+
+    with open("%s.argo_metrics.json" % (args.sampleId), 'w') as f:
+        f.write(json.dumps(mqc_stats_updated, indent=2))
```

Comment:
Same as above.
Change file pattern to contain more details to avoid the chance of file being overwritten when the same folder contains different types of metric files for the same sample.
For example: "%s.rnaaln_star_metrics.json" or "%s.rnaaln_hisat2_metrics.json". Will we be able to get the info of whether it is star of hisat2?